photo-upload.1
.TH "PHOTO-UPLOAD" "1" "February 15, 2014" "0.12" "Photo Uploader"
.SH NAME
photo-upload \- tool for uploading photos
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.\" Man page generated from reStructeredText.
.
.SH SYNOPSIS
.sp
.nf
.ft C
photo\-upload [options] images
.ft P
.fi
.SH DESCRIPTION
.sp
This manual page explains the photo\-upload program. This program is simple
uploader of photos to minilab.
.SH OPTIONS
.INDENT 0.0
.TP
.B \-\-version
Show program\(aqs version number and exit.
.UNINDENT
.INDENT 0.0
.TP
.B \-h, \-\-help
Show help message and exit.
.UNINDENT
.INDENT 0.0
.TP
.B \-\-license
Display program license.
.UNINDENT
.INDENT 0.0
.TP
.B \-s SERVICE_NAME, \-\-service=SERVICE_NAME
Name of service to use. Use photo\-upload \-l to list currently
available services.
.UNINDENT
.INDENT 0.0
.TP
.B \-l, \-\-list\-services
List available services.
.UNINDENT
.INDENT 0.0
.TP
.B \-b, \-\-open\-browser
Open order in browser after uploading.
.UNINDENT
.INDENT 0.0
.TP
.B \-B, \-\-use\-browser
Define browser to use when opening web page. Default is autodetected by python
webbrowser. You can also specify any parameter to browser, the URL will be
just appended to string you enter here and passed to shell.
.UNINDENT
.INDENT 0.0
.TP
.B \-d, \-\-debug
Show debugging output. This includes HTML files received from service.
.UNINDENT
.INDENT 0.0
.TP
.B \-S SESSION, \-\-session=SESSION
Existing session to reuse (some services won\(aqt work
without existing session).
.UNINDENT
.SH EXAMPLES
.sp
Uploading all photos from /folder to default service and opening web browser to finish order:
.sp
.nf
.ft C
photo\-upload \-b /folder/*.jpg
.ft P
.fi
.sp
Adding one image to existing session on ilikephoto.cz service:
.sp
.nf
.ft C
photo\-upload \-s ilikephoto.cz \-S f1721a19cc3c95218525a8429d48dab2 /tmp/Obraz024.jpg
.ft P
.fi
.sp
Listing currently supported services:
.sp
.nf
.ft C
photo\-upload \-l
.ft P
.fi
.sp
Uploading photo and opening epiphany to finish order:
.sp
.nf
.ft C
photo\-upload \-b \-B epiphany image.jpg
.ft P
.fi
.SH FILES
.SS \fB~/.photo\-upload\fP
.sp
Configuration file where default options are read. This is standard ini like
format file. You can use same values as long options, just specify them in
[photo\-upload] section. Some services might also use this config file, check
their documentation for details. Example configuration file:
.sp
.nf
.ft C
[photo\-upload]

service = droxi.cz
.ft P
.fi
.sp
Some services require additional configuration sections, eg. with login names:
.sp
.nf
.ft C
[happyfoto.cz]
user = name
password = passphrase
.ft P
.fi
.sp
Some services may require specifying registration code instead of the password:
.sp
.nf
.ft C
[imageshack.us]
user = name
regcode = 575af6686afe6a0c1d
.ft P
.fi
.sp
Imageshack also allows you to choose if the image is private or public, defaulting
to private. To change that, specify public = yes under imageshack.us section.
.SH LICENCE
.sp
This program is licensed under GNU/GPL version 2.
.SH BUGS
.sp
There are definitely many bugs, reporting to author is welcome. Please include
some useful information when sending bug reports (eg. exception you received
and debug output). Please submit your reports to <\fI\%http://bugs.cihar.com/\fP>.
.SH SEE ALSO
.sp
More information is available on program website:
<\fI\%http://cihar.com/software/photo-uploader/\fP>.
.SH AUTHOR
Michal Čihař <michal@cihar.com>
.SH COPYRIGHT
2007 - 2010, Michal Čihař
.\" Generated by docutils manpage writer.
.\" 
.


NEWS
NEWS
====

Major changes for each release are described here. Detailed changes between
two versions can be retrieved from Subversion.

0.6
=== (2007-??-??)

* Added support for iTesco.cz.

0.5
=== (2007-12-11)

* Added generic helper for ILikePhoto service.
* Added some error checking.
* Fixed handling of wrong parameters passed (non files).

0.4
=== (2007-11-07)

* Fixed crash.
* Added uloz.to service.

0.3
=== (2007-10-08)

* It is possible to get session ID from API.
* Added support for opening in user defined browser.
* Now parses ~/.photo-upload and uses it as defaults.
* Support for uploading to imageshack.us.

0.2
=== (2007-10-01)

* Modularize the code.
* Add support for new services.

0.1
=== (2007-09-25)

* Initial relase.



photo-upload.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
# vim: expandtab sw=4 ts=4 sts=4:
'''
Photo uploader, main command line script.
'''
__author__ = 'Michal Čihař'
__email__ = 'michal@cihar.com'
__license__ = '''
Copyright © 2007 - 2010 Michal Čihař

This program is free software; you can redistribute it and/or modify it
under the terms of the GNU General Public License version 2 as published by
the Free Software Foundation.

This program is distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
more details.

You should have received a copy of the GNU General Public License along with
this program; if not, write to the Free Software Foundation, Inc.,
51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
'''

from optparse import OptionParser
import phoupl
import webbrowser
import sys
import os

CONFIGFILE = '~/.photo-upload'
CONFIGDEFAULTS = {
        'photo-upload' : {
            'service' : 'postimage.org',
            'open-browser' : False,
            'use-browser' : '',
            },
        'imageshack.us' : {
            'public' : False,
            }
        }

def configure():
    '''
    Parameters and config file processing.
    '''
    config = phoupl.config.Config(CONFIGFILE, CONFIGDEFAULTS)

    program_name = 'photo-upload %s' % phoupl.__version__
    usage = "usage: %prog [options] images"
    parser = OptionParser(usage = usage, version = program_name)
    parser.add_option("", "--license",
                      action="store_true",
                      dest="show_license", default=False,
                      help="Display program license.")
    parser.add_option("-s", "--service",
                      action="store", type="string",
                      dest="service_name",
                      default=config.get('photo-upload', 'service'),
                      help="Name of service to use.")
    parser.add_option("-l", "--list-services",
                      action="store_true",
                      dest="list_services", default=False,
                      help="List available services.")
    parser.add_option("-b", "--open-browser",
                      action="store_true",
                      dest="open_browser",
                      default=config.getbool('photo-upload', 'open-browser'),
                      help="Open order in browser after uploading.")
    parser.add_option("-B", "--use-browser",
                      action="store",
                      dest="use_browser",
                      default=config.get('photo-upload', 'use-browser'),
                      help="Define browser to use when opening web page. " +
                      "Default is autodetected by python webbrowser.")
    parser.add_option("-d", "--debug",
                      action="store_true",
                      dest="debug", default=False,
                      help="Show debugging output.")
    parser.add_option("-S", "--session",
                      action="store", type="string",
                      dest="session", default=None,
                      help="Existing session to reuse "+
                      "(some services won't work without existing session).")

    (options, args) = parser.parse_args()

    # Informational options
    if options.show_license:
        print(program_name)
        print(__license__)
        sys.exit(0)

    return (options, args, parser, config)

def print_str(info, name):
    '''
    Prints single string value.
    '''
    try:
        print('%s: %s' % (name, info[name]))
    except KeyError:
        return

def print_list(info, name):
    '''
    Prints list value.
    '''
    try:
        print('%s: %s' % (name, ', '.join(info[name])))
    except KeyError:
        return

def list_services():
    '''
    List available services.
    '''
    for service in phoupl.list_services():
        print('Name: %s' % service)
        info = phoupl.get_service_info(service)
        print_str(info, 'FullName')
        print_str(info, 'Type')
        print_str(info, 'URL')
        print_list(info, 'Features')
        print_list(info, 'Languages')
        print_list(info, 'Countries')
        print()

def main():
    '''
    Main script.
    '''
    (options, args, parser, config) = configure()

    if options.debug:
        sys.stderr.write('Started photo-uploader, params:\n')
        sys.stderr.write('%s\n' % repr(sys.argv))
        sys.stderr.write('Options:\n')
        sys.stderr.write('%s\n' % repr(options))
        sys.stderr.write('Args:\n')
        sys.stderr.write('%s\n' % repr(args))

    # List services
    if options.list_services:
        list_services()
        sys.exit(0)

    # Check for some files to upload
    if len(args) == 0:
        parser.print_help()
        sys.exit("No files to upload.")

    urls = phoupl.upload_photos(options.service_name, args,
            debug = options.debug,
            session = options.session,
            configstorage = config)

    if type(urls) == str:
        urls = [urls]

    for url in urls:
        if options.open_browser:
            if options.use_browser != '':
                os.system('%s \'%s\' &' % (options.use_browser, url))
            else:
                webbrowser.open(url)


if __name__ == '__main__':
    main()
